### DISCUSSION TITLE
Error while hydrating message is unhelpful

### DISCUSSION THREAD

**Discussion by ThePaulMcBride** [vercel/next.js#42952] [Help]
I've got an issue where I'm getting the error while hydrating but the error message doesn't point me to where the issue is actually happening. This has happened on a few separate projects I've worked on.

This is what it looks like

<img width="961" alt="Screenshot 2022-11-15 at 15 16 06" src="https://user-images.githubusercontent.com/6891650/201956584-fe8f9d0b-8c2a-47a8-b74a-6d6576115560.png">

Is there a way I can track down the components causing this?

**Comment by icyJoseph** [comment]
This error is raised by react-dom, and it means that a bunch of things could've happen. 

For most cases, the console does print what went wrong. Have you inspected there? It usually says something like `expected X found Y`.

There's a kind of more cryptic way in which this could happen and that is invalid DOM nesting, but even then in some cases you'll see proper logs pointing you where to look for.

Other ways in which this can happen are when the server errors out, but the client succeeds to render, or vice-versa. 

Got any logs or repository to look at? I've seen so many of these I think I can actually help out with a bit more info. 

One strategy that usually works is to place a debugger at any of the functions called on the stack and try to see what nodes react-dom is comparing.

Though I am somehow suspicious of that sentryWrapped call, but let's not make big assumptions just yet.

**Reply by ThePaulMcBride** [reply]
The repo unfortunately I can't share, but it its a Next app on the most recent version using the most recent react etc too. We are using the Sentry error tracking plugin for next and I was able to catch this error.

https://sentry.io/share/issue/a6a8271470c443e4a82abd08968964bb/

Is that any use for debugging?

**Reply by icyJoseph** [reply]
It was not, I couldn't find the underlying error code, in what path are you seeing this error? I've been trying to figure out where it might be but I don't see the error on the console, perhaps because Sentry catches it and sends it over to your dashboard?
